api/src/main/java/org/openmrs/parameter/MedicationDispenseCriteria.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.parameter;
+
+import org.openmrs.Concept;
+import org.openmrs.Drug;
+import org.openmrs.DrugOrder;
+import org.openmrs.Encounter;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.Provider;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * The search parameter object for MedicationDispense. A convenience interface for building
+ * instances is provided by {@link MedicationDispenseCriteriaBuilder}.
+ * @since 2.6.0
+ * @see MedicationDispenseCriteriaBuilder
+ */
+public class MedicationDispenseCriteria {
+	
+	private Patient patient;
+	private Encounter encounter;
+	private DrugOrder drugOrder;
+	private boolean includeVoided = false;
+	
+	// TODO: Consider what else we might want to add here to start out with, eg:
```

Comment:
@mogoodrich - question here.  We can leave all of these off for now, I just added them in as examples of ways we might want to search and filter the data.  Let's look at the use cases presented by the dispensing UI and consider what makes the most sense to support as search criteria upfront.

Reply:
Yeah, makes sense... I believe I did intentionally derive the search specs from the basic use cases for Dispense that I knew we had so far, with the understanding that I inevitably would miss some.  In the UI design we have now, I believe all high-level searching happens at the Order/MedicationRequest level.  I don't think there's a case yet where we search for dispense events outside of orders.

Reply:
I like this having a criteria object pattern. Seems like it could save us quite a few override methods going forward.